Sources/TinyStorage/TinyStorage.swift

```diff
@@ -179,7 +179,7 @@ public final class TinyStorage: @unchecked Sendable {
     /// 4. This `migrate` function does not support nested collections due to Swift not having any `AnyCodable` type and the complication in supporting deeply nested types. That means `[String: Any]` is fine, provided `Any` is not another array or dictionary. The same applies to Arrays, `[String]` is okay but `[[String]]` is not. This includes arrays of dictionaries. This does not mean `TinyStorage` itself does not support nested collections (it does), however the migrator does not. You are still free to migrate these types manually as a result (in which case look at the `bulkStore` function).
     /// 5. As TinyStorage does not support mixed collection types, neither does this `migrate` function. For instance an array of `[Any]` where `Any` could be a `String` or `Int` is invalid, as is `[String: Any]` where `Any` is not one consistent type.
     public func migrate(userDefaults: UserDefaults, keys: Set<String>, overwriteIfConflict: Bool) {
-        dispatchQueue.sync {
+        dispatchQueue.async(flags: .barrier) { [self] in
```

Comment:
Not sure if this will be possible while also building for Swift 6, as `UserDefaults` does not conform to `Sendable`

I read a semi-related discussion here:
https://forums.developer.apple.com/forums/thread/757527